src/bci_build/package/basecontainers.py

```diff
@@ -84,42 +84,83 @@
     for os_version in ALL_BASE_OS_VERSIONS
 ]
 
+_FIPS_ASSET_BASEURL = "https://api.opensuse.org/public/build/"
+
 # https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp3991.pdf
 # Chapter 9.1 Crypto Officer Guidance
-_FIPS_15_SP2_ASSET_BASEURL = "https://api.opensuse.org/public/build/"
-_FIPS_15_SP2_BINARIES = [
+_FIPS_15_SP2_BINARIES: list[str] = [
     f"SUSE:SLE-15-SP2:Update/pool/x86_64/openssl-1_1.18804/{name}-1.1.1d-11.20.1.x86_64.rpm"
     for name in ("openssl-1_1", "libopenssl1_1", "libopenssl1_1-hmac")
 ] + [
     f"SUSE:SLE-15-SP1:Update/pool/x86_64/libgcrypt.15117/{name}-1.8.2-8.36.1.x86_64.rpm"
     for name in ("libgcrypt20", "libgcrypt20-hmac")
 ]
 
+# submitted, not yet certified
+_FIPS_15_SP4_BINARIES: list[str] = [
+    f"SUSE:SLE-15-SP4:Update/pool/x86_64/openssl-1_1.28168/{name}-1.1.1l-150400.7.28.1.x86_64.rpm"
+    for name in ("openssl-1_1", "libopenssl1_1", "libopenssl1_1-hmac")
+] + [
+    f"SUSE:SLE-15-SP4:Update/pool/x86_64/libgcrypt.28151/{name}-1.9.4-150400.6.8.1.x86_64.rpm"
+    for name in ("libgcrypt20", "libgcrypt20-hmac")
+]
+
+# see grep -c -E "^ *V3.*key ID 39db7c82: OK" *.rpm | wc -l  below
```

Comment:
I'm afraid I'm too dumb to understand this comment

Reply:
I have two arrays depending on the SP, but a "len" check later in the shell script (that was a requirement from autobuild to ensure that no unsigned packages are injected in the build). I didn't want to repeat the version condition there so I just added the assert and always use the SP2 len. it's not super important as when the array changes length the condition will fail.